lib/otel/span-processor.js

```diff
@@ -62,19 +64,34 @@ module.exports = class NrSpanProcessor {
   }
 
   /**
-   * Update the segment duration from span and reconcile
-   * any attributes that were added after the start
+   * Update the segment duration from span, handle errors and
+   * reconcile any attributes that were added after the start
    * @param {object} span otel span getting updated
    */
   onEnd(span) {
     if (span[otelSynthesis] && span[otelSynthesis].segment) {
       const { segment, transaction } = span[otelSynthesis]
       this.updateDuration(segment, span)
+      if (span?.status?.code === SPAN_STATUS_CODE.ERROR) {
+        this.handleError(segment, transaction, span)
+      }
       this.reconcileAttributes({ segment, span, transaction })
       delete span[otelSynthesis]
     }
   }
 
+  handleError(segment, transaction, span) {
+    const errorEvents = span.events.filter((event) => event.name === 'exception')
+    errorEvents.forEach((err) => {
+      if (err) {
```

Comment:
why would this be falsey?

Reply:
I think that was left in there accidentally after our pairing. I'll remove it.